src/pages/Question.tsx

```diff
@@ -40,79 +38,81 @@ const MOVIES = gql`
   }
 `
 
+const GET_SCORE = gql`
+  query($imdbId: String!, $selectedTitle: String!, $remainingSeconds: Int!) {
+    scoreTitleResponse(
+      imdbId: $imdbId
+      selectedTitle: $selectedTitle
+      remainingSeconds: $remainingSeconds
+    )
+  }
+`
+
 const Question = () => {
   const history = useHistory()
 
-  useStorage()
-
-  const { addPoints, removeLife, currentImdbId, timeRemaining } = useStore()
+  const {
+    addPoints,
+    removeLife,
+    currentImdbId,
+    timeRemaining,
+    setTimeRemaining,
+  } = useStore()
 
   const { loading, error, data } = useQuery<MovieData>(MOVIES, {
-    variables: { imdbId: currentImdbId },
+    fetchPolicy: 'cache-only',
   })
 
-  const [movies, setMovies] = useState<Movie[]>([])
-
-  useIonViewWillEnter(() => {
-    if (data) {
-      const movie: Movie = {
-        imdbId: data.movie.imdbId,
-        title: data.movie.title,
-        releaseYear: data.movie.releaseYear,
-        posterPath: data.movie.posterPath,
-        randomMovies: [],
-      }
-
-      const randomMovies: Movie[] = [...data.movie.randomMovies]
-
-      // combine all movie object into one array
-      const _movies = [movie, ...randomMovies]
-
-      setMovies(randomArrayShuffle(_movies))
-    }
-  }, [data])
+  const [submitResponse, scoreResponse] = useLazyQuery(GET_SCORE)
 
-  const navigateSuccess = () => {
-    history.replace('/success')
-  }
-
-  const navigateFailure = () => {
-    removeLife()
-    history.replace('/failure')
-  }
-
-  const handleMovieSelect = async (movieTitle: string) => {
-    const score = await movieService.validateAnswer(
-      movieTitle,
-      currentImdbId,
-      timeRemaining,
-    )
-
-    if (score === 0) {
-      navigateFailure()
-    }
+  // create a list of all movies that should be shuffled (but only once)
+  const allMovies = useMemo(
+    () =>
+      randomArrayShuffle([...(data?.movie?.randomMovies || []), data?.movie]),
+    [data],
+  )
 
-    if (score && score > 0) {
-      addPoints(score)
-      navigateSuccess()
+  // create an effect that is executed on changes to scoreResponse in out lazy query
+  // => will be called once on first render and once when a scoring response is received
+  useEffect(() => {
+    if (Number.isFinite(scoreResponse?.data?.scoreTitleResponse)) {
```

Comment:
Cannot simply check for the existence of scoreTitleResponse, as having it be 0 would fail the if condition